DependencyInjection/KnpMenuExtension.php

```diff
@@ -38,10 +40,45 @@ public function load(array $configs, ContainerBuilder $container)
         }
 
         $container->setParameter('knp_menu.default_renderer', $config['default_renderer']);
+
+        if ($menus = $config['menus']) {
+            $loader->load('config_provider.xml');
+
+            if (class_exists('Symfony\Component\ExpressionLanguage\ExpressionLanguage')) {
+                $loader->load('expression.xml');
+                $menus = $this->preParseExpressions($menus, $container);
+            }
+
+            $container->getDefinition('knp_menu.menu_provider.config')->replaceArgument(0, $menus);
+        }
     }
 
     public function getNamespace()
     {
         return 'http://knplabs.com/schema/dic/menu';
     }
+
+    private function preParseExpressions(array $menus, ContainerBuilder $container)
+    {
+        $contextClass = $container->getParameter('knp_menu.expression_context.class');
```

Comment:
This will always be the same class here, because nothing has a chance to change the parameter between the loading of the file and the call to ``preParseExpressions``. You are making the code more complex for nothing.